File: CentralDeAjuda-Back/src/main/java/uniamerica/centralDeAjuda/Entity/Auditoria.java
package uniamerica.centralDeAjuda.Entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter @Setter
//@MappedSuperclass
@Entity
public class Auditoria {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "aluno_id")
    private Aluno aluno;
    @ManyToOne
    @JoinColumn(name = "notebook_id")
    private Notebook notebook;
    @ManyToOne
    @JoinColumn(name = "marca_id")
    @JsonIgnoreProperties("modelos")
    private Marca marca;
    @ManyToOne
    @JoinColumn(name = "modelo_id")
    private Modelo modelo;
    @ManyToOne
    @JoinColumn(name = "ticket_id")
    private Ticket ticket;
    private String userCriacao;
    private String userAlteracao;
    private String userExclusao;
    private String userFinalizacao;
    private LocalDateTime dataHoraCriacao;
    private LocalDateTime dataHoraAlteracao;
    private LocalDateTime dataHoraExclusao;
    private LocalDateTime dataHoraFinalizacao;

}

File: CentralDeAjuda-Front/src/app/models/auditoria.ts
import { Aluno } from "./aluno";
import { Marca } from "./marca";
import { Modelo } from "./modelo";
import { Notebook } from "./notebook";
import { Ticket } from "./ticket";

export class Auditoria {
    id!: number;
    aluno!: Aluno;
    notebook!: Notebook;
    marca!: Marca;
    modelo!: Modelo;
    ticket!: Ticket;
    userCriacao!: string;
    userAlteracao!: string;
    userExclusao!: string;
    dataHoraCriacao!: Date;
    dataHoraAlteracao!: Date;
    dataHoraExclusao!: Date;
}